unittests/api_tests.cpp

```diff
@@ -2372,6 +2372,38 @@ BOOST_FIXTURE_TEST_CASE(action_ordinal_test, TESTER) { try {
    set_code( N(erin), contracts::test_api_wasm() );
    produce_blocks(1);
 
+   // generate test hash
+   auto gth = [&](auto ac) {
+      action testact;
+      testact.account = ac.account;
+      testact.name    = ac.name;
+      testact.authorization = ac.authorization;
+      std::vector<char> buff(fc::raw::pack_size(testact));
+      datastream<char*> ds(buff.data(), buff.size());
+      fc::raw::pack(ds, testact);
+      sha256::encoder enc;
+      enc.write(buff.data(), fc::raw::pack_size(*(action_base*)(&testact)));
+      auto ret = enc.result();
+      return ret;
+   };
+
+   // prove act digest
+   auto pad = [&](const auto& action_hash, const auto& test_hash, auto payload, auto value) {
+      std::vector<char> buff(1 + fc::raw::pack_size(value));
+      datastream<char*> ds(buff.data(), buff.size());
+      fc::raw::pack(ds, std::vector<char>{});
+      fc::raw::pack(ds, value);
```

Comment:
Wait, isn't this packing an `optional<vector<char>>`? Again, how do these tests even pass?

Reply:
Okay, I see why they pass: https://github.com/EOSIO/eos/pull/8737#discussion_r390699583

This test function is also reproducing the buggy hash calculation.